src/Kobo/SystemDialog.cpp

```diff
@@ -56,11 +58,15 @@ void
 SystemWidget::Prepare(ContainerWindow &parent, const PixelRect &rc) noexcept
 {
   AddButton("Reboot", [](){ KoboReboot(); });
-  AddButton(IsKoboOTGKernel() ? "Disable USB-OTG" : "Enable USB-OTG",
-            [this](){ SwitchKernel(); });
-
+  switch_kernel_button = AddButton(IsKoboOTGKernel() ?
+                                   "Disable USB-OTG" : "Enable USB-OTG",
+                                   [this](){ SwitchKernel(); });
   AddButton("Export USB storage", [this](){ ExportUSBStorage(); });
   SetRowEnabled(USB_STORAGE, !IsKoboOTGKernel());
+#ifdef KOBO
+  if (DetectKoboModel() == KoboModel::CLARA_HD)
+    switch_kernel_button->SetEnabled(0);
+#endif
```

Comment:
Too complicated. Look at the existing code line above that - you can simply call `SetRowEnabled()`, and then you don't even need the variable.
Also note that `0` is not a `bool`. It gets casted implicitly to `bool`, but it's the wrong parameter type.